papier/README.md

```diff
@@ -0,0 +1,149 @@
+as a custom metric for Arm MAP.  It enables MAP to calculate 
+arbitrary mathematical expressions involving data gathered from 
+multiple measurement systems, e.g. perf, PAPI, CUPTI, or the OS.
+
+Quick Start
+-----------
+
+ * Edit Makefile as needed and `make install`
+ * `export ALLINEA_PAPIER_EXPRESSION="(PAPI{PAPI_FP_INS} + PAPI{PAPI_VEC_INS}) / PAPI{PAPI_TOT_INS}"`
+ * `map [options] ./a.out`
+
+Usage
+-----
+
+Set the `ALLINEA_PAPIER_EXPRESSION` environment variable to 
+the expression itself or a path to a file containing the 
+expression.  Once the custom metric is installed you won't have 
+to recompile or write code or anything to get new data.  Just 
+set an environment variable and you're done!
+
+The syntax for specifying metrics is `<SYSTEM>{METRIC}`, e.g. 
+PAPI{PAPI_L1_TCM} or TIME{WALLCLOCK} or PAPI{perf_raw::r0156}.
+See "Measurement Systems" below for more details.
+
+The front end is based on [ExprTk][1], so performance metrics 
+can be defined with \rich mathematical expressions e.g. these 
+are all valid expressions (copied from the ExprTk readme): 
+
+ - `sqrt(1 - (3 / x^2))`
+ - `clamp(-1, sin(2 * pi * x) + cos(y / 2 * pi), +1)`
+ - `sin(2.34e-3 * x)`
+ - `if(((x[2] + 2) == 3) and ((y + 5) <= 9),1 + w, 2 / z)`
+ - `inrange(-2,m,+2) == if(({-2 <= m} and [m <= +2]),1,0)`
+ - `({1/1}*[1/2]+(1/3))-{1/4}^[1/5]+(1/6)-({1/7}+[1/8]*(1/9))`
+ - `a * exp(2.2 / 3.3 * t) + c`
+ - `z := x + sin(2.567 * pi / y)`
+ - `u := 2.123 * {pi * z} / (w := x + cos(y / pi))`
+ - `2x + 3y + 4z + 5w == 2 * x + 3 * y + 4 * z + 5 * w`
+ - `3(x + y) / 2.9 + 1.234e+12 == 3 * (x + y) / 2.9 + 1.234e+12`
+ - `(x + y)3.3 + 1 / 4.5 == [x + y] * 3.3 + 1 / 4.5`
+ - `(x + y[i])z + 1.1 / 2.7 == (x + y[i]) * z + 1.1 / 2.7`
+ - `(sin(x / pi) cos(2y) + 1) == (sin(x / pi) * cos(2 * y) + 1)`
+ - `75x^17 + 25.1x^5 - 35x^4 - 15.2x^3 + 40x^2 - 15.3x + 1`
+ - `(avg(x,y) <= x + y ? x - y : x * y) + 2.345 * pi / x`
+ - `while (x <= 100) { x -= 1; }`
+ - `x <= 'abc123' and (y in 'AString') or ('1x2y3z' != z)`
+ - `((x + 'abc') like '*123*') or ('a123b' ilike y)`
+ - `sgn(+1.2^3.4z / -5.6y) <= {-7.8^9 / -10.11x }`
+ 
+On each sample, Papier updates variables (e.g. x, y, z, u, etc.) 
+with performance data.  Conditionals, trans, trig... go wild!
+
+
+Available Metrics
+-----------------
+
+The backend can pull data from any source defined as a subclass of
+the `Metrics` class.  This release supports:
+
+ * [PAPI][2]: Hardware performance counters.
+   * Any PAPI predefined, native, or raw perf metric.  
+     See `papi_avail` and `papi_native_avail`.
+ * TIME: Timers
+   * WALLCLOCK: application wallclock time.
+   * CPU: CPU time
+ * ALLINEA: Metrics gathered from or related to the Allinea MAP tool.
+   * SAMPLE_COUNT: Number of samples taken so far.
+   * LOGICAL_CORES: Logical cores as reported by MAP.
+   * PHYSICAL_CORES: Physical cores as reported by MAP.
+
+
+Examples
+--------
+
+Tested on the Marvel ThunderX2 with PAPI 5.7.0.1.
```

Comment:
Please just say which version of PAPI you tested with and remove the reference to ThunderX 2.